rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -367,11 +369,13 @@ Node::list_parameters(
     {
       result.names.push_back(kv.first);
       size_t last_separator = kv.first.find_last_of('.');
-      std::string prefix = kv.first.substr(0, last_separator);
-      if (std::find(result.prefixes.cbegin(), result.prefixes.cend(),
-        prefix) == result.prefixes.cend())
-      {
-        result.prefixes.push_back(prefix);
+      if (std::string::npos != last_separator) {
+        std::string prefix = kv.first.substr(0, last_separator);
+        if (std::find(result.prefixes.cbegin(), result.prefixes.cend(),
+          prefix) == result.prefixes.cend())
```

Comment:
I also had trouble with that line, but I wasn't sure what to suggest at the time. I think that this is only slightly more readable:

```c++
        if (std::find(result.prefixes.cbegin(), result.prefixes.cend(), prefix) ==
          result.prefixes.cend())
        {
          /* stuff... */
        }
```